File: invesdwin-context-matlab-parent/invesdwin-context-matlab-runtime-contract/src/test/java/de/invesdwin/context/matlab/runtime/contract/InputsAndResultsTestString.m
disp('getString')
if exist('getString')
	error('getString already defined!')
end
getString = putString
disp(class(getString))
disp(getString)
if ~isa(getString, 'char')
	error('getString not char!')
end

disp('getStringWithNull')
if exist('getStringWithNull')
	error('getStringWithNull already defined!')
end
getStringWithNull = putStringWithNull
disp(class(getStringWithNull))
disp(getStringWithNull)
if ~isa(getStringWithNull, 'double')
	error('getStringWithNull not double!')
end
if ~isnan(getStringWithNull)
	error('getStringWithNull not nan!')
end

disp('getStringVector')
if exist('getStringVector')
	error('getStringVector already defined!')
end
getStringVector = putStringVector
disp(class(getStringVector))
disp(getStringVector)
if ~isa(getStringVector, 'cell')
	error('getStringVector not cell!')
end


disp('getStringVectorWithNull')
if exist('getStringVectorWithNull')
	error('getStringVectorWithNull already defined!')
end
getStringVectorWithNull = putStringVectorWithNull
disp(class(getStringVectorWithNull))
disp(getStringVectorWithNull)
if ~isa(getStringVectorWithNull, 'cell')
	error('getStringVectorWithNull not cell!')
end
if ~isempty(getStringVectorWithNull{2})
	error('getStringVectorWithNull{2} not empty!')
end

disp('getStringVectorAsList')
if exist('getStringVectorAsList')
	error('getStringVectorAsList already defined!')
end
getStringVectorAsList = putStringVectorAsList
disp(class(getStringVectorAsList))
disp(getStringVectorAsList)
if ~isa(getStringVectorAsList, 'cell')
	error('getStringVectorAsList not cell!')
end

disp('getStringVectorAsListWithNull')
if exist('getStringVectorAsListWithNull')
	error('getStringVectorAsListWithNull already defined!')
end
getStringVectorAsListWithNull = putStringVectorAsListWithNull
disp(class(getStringVectorAsListWithNull))
disp(getStringVectorAsListWithNull)
if ~isa(getStringVectorAsListWithNull, 'cell')
	error('getStringVectorAsListWithNull not cell!')
end
if ~isempty(getStringVectorAsListWithNull{2})
	error('getStringVectorAsListWithNull{2} not empty!')
end

disp('getStringMatrix')
if exist('getStringMatrix')
	error('getStringMatrix already defined!')
end
getStringMatrix = putStringMatrix
disp(class(getStringMatrix))
disp(getStringMatrix)
if ~isa(getStringMatrix, 'cell')
	error('getStringMatrix not cell!')
end


disp('getStringMatrixWithNull')
if exist('getStringMatrixWithNull')
	error('getStringMatrixWithNull already defined!')
end
getStringMatrixWithNull = putStringMatrixWithNull
disp(class(getStringMatrixWithNull))
disp(getStringMatrixWithNull)
if ~isa(getStringMatrixWithNull, 'cell')
	error('getStringMatrixWithNull not cell!')
end
if ~isempty(getStringMatrixWithNull{1,1})
	error('getStringMatrixWithNull{1,1} not empty!')
end
if ~isempty(getStringMatrixWithNull{2,2})
	error('getStringMatrixWithNull{2,2} not empty!')
end
if ~isempty(getStringMatrixWithNull{3,3})
	error('getStringMatrixWithNull{3,3} not empty!')
end

disp('getStringMatrixAsList')
if exist('getStringMatrixAsList')
	error('getStringMatrixAsList already defined!')
end
getStringMatrixAsList = putStringMatrixAsList
disp(class(getStringMatrixAsList))
disp(getStringMatrixAsList)
if ~isa(getStringMatrixAsList, 'cell')
	error('getStringMatrixAsList not cell!')
end

disp('getStringMatrixAsListWithNull')
if exist('getStringMatrixAsListWithNull')
	error('getStringMatrixAsListWithNull already defined!')
end
getStringMatrixAsListWithNull = putStringMatrixAsListWithNull
disp(class(getStringMatrixAsListWithNull))
disp(getStringMatrixAsListWithNull)
if ~isa(getStringMatrixAsListWithNull, 'cell')
	error('getStringMatrixAsListWithNull not cell!')
end
if ~isempty(getStringMatrixAsListWithNull{1,1})
	error('getStringMatrixAsListWithNull{1,1} not empty!')
end
if ~isempty(getStringMatrixAsListWithNull{2,2})
	error('getStringMatrixAsListWithNull{2,2} not empty!')
end
if ~isempty(getStringMatrixAsListWithNull{3,3})
	error('getStringMatrixAsListWithNull{3,3} not empty!')
end

File: invesdwin-context-matlab-parent/invesdwin-context-matlab-runtime-javasci/src/test/java/de/invesdwin/context/matlab/runtime/javasci/test/InputsAndResultsTestString.java
package de.invesdwin.context.matlab.runtime.javasci.test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import javax.annotation.concurrent.NotThreadSafe;

import org.springframework.core.io.ClassPathResource;

import de.invesdwin.context.integration.script.IScriptTaskEngine;
import de.invesdwin.context.integration.script.IScriptTaskInputs;
import de.invesdwin.context.integration.script.IScriptTaskResults;
import de.invesdwin.context.matlab.runtime.contract.AScriptTaskMatlab;
import de.invesdwin.context.matlab.runtime.contract.IScriptTaskRunnerMatlab;
import de.invesdwin.util.assertions.Assertions;
import de.invesdwin.util.lang.Objects;

@NotThreadSafe
public class InputsAndResultsTestString {

    private final IScriptTaskRunnerMatlab runner;

    public InputsAndResultsTestString(final IScriptTaskRunnerMatlab runner) {
        this.runner = runner;
    }

    public void testString() {
        //putString
        final String putString = "asdf";
        final String putStringWithNull = null;

        //putStringVector
        final String[] putStringVector = new String[3];
        for (int i = 0; i < putStringVector.length; i++) {
            putStringVector[i] = i + "-" + i;
        }
        final String[] putStringVectorWithNull = Objects.deepClone(putStringVector);
        putStringVectorWithNull[1] = null;

        //putStringVectorAsList
        final List<String> putStringVectorAsList = Arrays.asList(putStringVector);
        final List<String> putStringVectorAsListWN = Objects.deepClone(putStringVectorAsList);
        putStringVectorAsListWN.set(1, null);

        //putStringMatrix
        final String[][] putStringMatrix = new String[4][];
        for (int i = 0; i < putStringMatrix.length; i++) {
            final String[] vector = new String[3];
            for (int j = 0; j < vector.length; j++) {
                vector[j] = i + "" + j + "-" + i + "" + j;
            }
            putStringMatrix[i] = vector;
        }
        final String[][] putStringMatrixWithNull = Objects.deepClone(putStringMatrix);
        for (int i = 0; i < putStringMatrixWithNull[0].length; i++) {
            putStringMatrixWithNull[i][i] = null;
        }

        //putStringMatrixAsList
        final List<List<String>> putStringMatrixAsList = new ArrayList<List<String>>(putStringMatrix.length);
        for (final String[] vector : putStringMatrix) {
            putStringMatrixAsList.add(Arrays.asList(vector));
        }
        final List<List<String>> putStringMatrixAsListWN = Objects.deepClone(putStringMatrixAsList);
        for (int i = 0; i < putStringMatrixAsListWN.get(0).size(); i++) {
            putStringMatrixAsListWN.get(i).set(i, null);
        }

        new AScriptTaskMatlab<Void>() {

            @Override
            public void populateInputs(final IScriptTaskInputs inputs) {
                inputs.putString("putString", putString);
                inputs.putString("putStringWithNull", putStringWithNull);

                inputs.putStringVector("putStringVector", putStringVector);
                inputs.putStringVector("putStringVectorWithNull", putStringVectorWithNull);

                inputs.putStringVectorAsList("putStringVectorAsList", putStringVectorAsList);
                inputs.putStringVectorAsList("putStringVectorAsListWN", putStringVectorAsListWN);

                inputs.putStringMatrix("putStringMatrix", putStringMatrix);
                inputs.putStringMatrix("putStringMatrixWithNull", putStringMatrixWithNull);

                inputs.putStringMatrixAsList("putStringMatrixAsList", putStringMatrixAsList);
                inputs.putStringMatrixAsList("putStringMatrixAsListWN", putStringMatrixAsListWN);
            }

            @Override
            public void executeScript(final IScriptTaskEngine engine) {
                engine.eval(new ClassPathResource(InputsAndResultsTestString.class.getSimpleName() + ".sce",
                        InputsAndResultsTestString.class));
            }

            @Override
            public Void extractResults(final IScriptTaskResults results) {
                //getString
                final String getString = results.getString("getString");
                Assertions.assertThat(putString).isEqualTo(getString);
                final String getStringWithNull = results.getString("getStringWithNull");
                Assertions.assertThat(putStringWithNull).isEqualTo(getStringWithNull);

                //getStringVector
                final String[] getStringVector = results.getStringVector("getStringVector");
                Assertions.assertThat(putStringVector).isEqualTo(getStringVector);
                final String[] getStringVectorWithNull = results.getStringVector("getStringVectorWithNull");
                Assertions.assertThat(putStringVectorWithNull).isEqualTo(getStringVectorWithNull);

                //getStringVectorAsList
                final List<String> getStringVectorAsList = results.getStringVectorAsList("getStringVectorAsList");
                Assertions.assertThat(putStringVectorAsList).isEqualTo(getStringVectorAsList);
                final List<String> getStringVectorAsListWN = results.getStringVectorAsList("getStringVectorAsListWN");
                Assertions.assertThat(putStringVectorAsListWN).isEqualTo(getStringVectorAsListWN);

                //getStringMatrix
                final String[][] getStringMatrix = results.getStringMatrix("getStringMatrix");
                Assertions.assertThat(putStringMatrix).isEqualTo(getStringMatrix);
                final String[][] getStringMatrixWithNull = results.getStringMatrix("getStringMatrixWithNull");
                Assertions.assertThat(putStringMatrixWithNull).isEqualTo(getStringMatrixWithNull);

                //getStringMatrixAsList
                final List<List<String>> getStringMatrixAsList = results.getStringMatrixAsList("getStringMatrixAsList");
                Assertions.assertThat(putStringMatrixAsList).isEqualTo(getStringMatrixAsList);
                final List<List<String>> getStringMatrixAsListWN = results
                        .getStringMatrixAsList("getStringMatrixAsListWN");
                Assertions.assertThat(putStringMatrixAsListWN).isEqualTo(getStringMatrixAsListWN);
                return null;
            }
        }.run(runner);
    }

}
